internal/server/data/grant.go

```diff
@@ -56,6 +57,34 @@ func ByOptionalPrivilege(s string) SelectorFunc {
 	}
 }
 
+func GrantsInheritedBySubject(subjectID uid.PolymorphicID) SelectorFunc {
+	return func(db *gorm.DB) *gorm.DB {
+		switch {
+		case subjectID.IsIdentity():
+			userID, err := subjectID.ID()
+			if err != nil {
+				logging.Errorf("invalid subject id %q", subjectID)
+				return db.Where("1 = 0")
+			}
+			var groupIDs []uid.ID
+			err = db.Session(&gorm.Session{Logger: db.Logger}).Raw("select distinct group_id from identities_groups where identity_id = ?", userID).Pluck("group_id", &groupIDs).Error
+			if err != nil {
+				logging.Errorf("GrantsInheritedByUser: %s", err)
```

Comment:
Using `db.Session` here looks correct, it's a separate query so needs to be a separate session.

Doing a separate independent query in a selector is a bit strange, but unless we change this to a join I don't see any way around it until we replace selectors. So I think this works for now.

Maybe we should `return db.Where("1 = 0")` here as well , like above?

Reply:
Looking at this again, I think we probably want `gorm.Session{NewDB: true}`. That should erase any previous statements added to the session by other selectors.